### PR TITLE
#1338 Always determine collection argument type when searching for adder

### LINKED ISSUE #1338: In protobuf3, how to mapping a ProtocolStringList
`ProtocolStringList` is `List<String>` in protobuf, mapstruct will not think this is a collection, because no type parameter.

```java
// this is a collection, so this can be done always
if ( !collectionProperty.getTypeParameters().isEmpty() ) {}
```
I need to tell mapstruct to call `addPropertyName` method, not `getPropertyNameList().addAll()`, so , add a mapping from `ProtocolStringList` to `List<String>` will not work.

Maybe mapstruct should support property based addAll method, like `addProperty`, so, mapstruct should use `addAllPropertyName( java.lang.Iterable<java.lang.String> values)`. Then all list in protobuf3 will just works.

### PR DESCRIPTION
**Pull request by filiphr**
If the Collection type is not actually generic it has no type parameters.
However, it's type argument can be determined.
One such list exists in protobuf (ProtocolStringList)

Fixes #1338

### CODE REVIEW

**Comment by sjaakd** [comment]
I'll have a look asp.

File: processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java [hunk 1]
```diff
@@ -560,23 +562,21 @@ private Accessor getAdderForType(Type collectionProperty, String pluralPropertyN
...
-                    ExecutableElement executable = adder.getExecutable();
-                    if ( executable == null ) {
-                        // it should not be null, but to be safe
-                        continue;
-                    }
-                    VariableElement arg = executable.getParameters().get( 0 );
-                    if ( typeUtils.isSameType( arg.asType(), typeArg ) ) {
-                        candidates.add( adder );
-                    }
+            TypeMirror typeArg = first( collectionProperty.determineTypeArguments( Iterable.class ) ).getTypeBound()
```
Comment (sjaakd) [inline processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java]: nice.. I think that the original solution predates the `determineTypeArguments`

Comment (filiphr) [reply processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java]: I think so too 😄

**Comment by filiphr** [comment]
Thanks for the review @sjaakd

**Pull request merged**